tests/tracking/test_rest_tracking.py

```diff
@@ -224,8 +224,7 @@ def test_log_metrics_params_tags(mlflow_client):
     experiment_id = mlflow_client.create_experiment('Oh My')
     created_run = mlflow_client.create_run(experiment_id)
     run_id = created_run.info.run_uuid
-    # TODO(sid): pass and assert on step
-    mlflow_client.log_metric(run_id, key='metric', value=123.456, timestamp=789)
+    mlflow_client.log_metric(run_id, key='metric', value=123.456, timestamp=789, step=2)
```

Comment:
Test logging a metric with a non-zero step using the REST client (this doesn't actually test new functionality added by this PR - the same test would have passed after we merged thee code changes in #1142)